internal/driver/driver.go

```diff
@@ -208,6 +212,25 @@ func applyCommandOverrides(cmd string, outputFormat int, cfg config) config {
 	return cfg
 }
 
+// generateTagRootsLeaves generates extra nodes from the tagroot and tagleaf options.
+func generateTagRootsLeaves(prof *profile.Profile, cfg config, ui plugin.UI) {
+	tagRootLabelKeys := withoutEmptyStrings(strings.Split(cfg.TagRoot, ","))
+	tagLeafLabelKeys := withoutEmptyStrings(strings.Split(cfg.TagLeaf, ","))
+	rootm, leafm := addLabelNodes(prof, tagRootLabelKeys, tagLeafLabelKeys, cfg.Unit)
+	warnNoMatches(cfg.TagRoot == "" || rootm, "TagRoot", ui)
+	warnNoMatches(cfg.TagLeaf == "" || leafm, "TagLeaf", ui)
+}
+
+// withoutEmptyStrings filters a slice to only non-empty strings
```

Comment:
nit: maybe dropEmptyStrings to start the function name with a verb.

Reply:
Done. I had drop to start with and switched it to 'without' because it was copying not mutating. I was probably overthinking it.